share/models/identifiers.py

```diff
@@ -0,0 +1,57 @@
+from furl import furl
+
+from django.db import models
+
+from share.models.base import ShareObject
+from share.models.fields import ShareForeignKey, ShareURLField
+
+__all__ = ('CreativeWorkIdentifier', 'PersonIdentifier')
+
+
+# TODO Common interface, so we're not duplicating code. Doesn't work because
+# ShareObjectMeta doesn't look at bases when building Version classes.
+#
+# class Identifier(ShareObject):
+#    # http://twitter.com/berniethoughts/, mailto://contact@cos.io
+#    uri = ShareURLField(unique=True)
+#
+#    # twitter.com, cos.io
+#    host = models.TextField(editable=False)
+#
+#    # http, mailto
+#    scheme = models.TextField(editable=False)
+#
+#    def save(self, *args, **kwargs):
+#        f = furl(self.uri)
+#        self.host = f.host
+#        self.scheme = f.scheme
+#        super(Identifier, self).save(*args, **kwargs)
+#
+#    class Meta:
+#        abstract = True
+
+
+class CreativeWorkIdentifier(ShareObject):
+    uri = ShareURLField(unique=True)
```

Comment:
We'll have to document fairly thoroughly what everything here is.
Should we switch to calling them irirs, authority, and scheme? I think that's how the RFC for IRIs refer to them. I personally thing uri is a touch more clear at first glance but I technically speaking uri's don't support unicode (I think that's the main difference)